sympy/polys/numberfields/modules.py

```diff
@@ -678,15 +702,30 @@ def __init__(self, T):
         Parameters
         ==========
 
-        T : :py:class:`~.Poly`
-            The monic, irreducible, univariate polynomial over :ref:`ZZ`, a
-            root of which is the generator of the power basis.
+        T : :py:class:`~.Poly`, :py:class:`~.AlgebraicField`
+            Either (1) the monic, irreducible, univariate polynomial over
+            :ref:`ZZ`, a root of which is the generator of the power basis,
+            or (2) an :py:class:`~.AlgebraicField` whose primitive element
+            is the generator of the power basis.
 
         """
+        K = None
+        if isinstance(T, AlgebraicField):
+            K, T = T, T.ext.minpoly_of_element()
+        if T.domain == QQ:
+            try:
+                T = Poly(T, domain=ZZ)
```

Comment:
The direct way to do this is `T.set_domain(ZZ)`.

Reply:
I guess it isn't necessary to check if the domain is `QQ` since it should also raise for e.g. `ZZ[x]` if it can't be coerced to `ZZ`.

Another possibility would be to use `clear_denoms` I guess.

Reply:
`set_domain()` looks good for this. Thanks, was not aware of this method!